test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -219,6 +226,167 @@ func (data *testData) pingBetweenInterfaces(t *testing.T) error {
+			}
+		}
+	]`)
+
+	if err := data.e2eTestData.PatchDaemonSet(namespace, name, patch); err != nil {
+		return fmt.Errorf("failed to pause antrea-agent %s: %v", name, err)
+	}
+	fmt.Println("Antrea Agent paused successfully")
+
+	logs.Infof("Deleting VLAN pod %s", data.pods[1].podName)
+	if err := data.e2eTestData.DeletePodAndWait(defaultTimeout, data.pods[1].podName, "default"); err != nil {
+		return fmt.Errorf("failed to delete pod %s: %v", data.pods[1].podName, err)
+	}
+
+	// resume antrea-agent pods
+	patch = []byte(`[
+		{
+			"op": "remove",
+			"path": "/spec/template/spec/nodeSelector"
+		}
+	]`)
+
+	if err := data.e2eTestData.PatchDaemonSet(namespace, name, patch); err != nil {
+		return fmt.Errorf("failed to resume Antrea agent: %v, output: %s", err, name)
+	}
+	fmt.Println("Antrea Agent resumed successfully")
+
+	return nil
+}
+
+func (data *testData) getPodPortsAndIPs(targetPod *testPodInfo) (map[string][]string, error) {
+	cmd := []string{"ip", "addr", "show"}
+	stdout, _, err := data.e2eTestData.RunCommandFromPod(data.e2eTestData.GetTestNamespace(), targetPod.podName, containerName, cmd)
+	if err != nil {
+		return nil, fmt.Errorf("error listing interfaces for pod %s: %w", targetPod.podName, err)
+	}
+
+	portsAndIPs := make(map[string][]string) // to store interface names and their associated IPs
+	var ifaceName string
+	for _, line := range strings.Split(stdout, "\n") {
+		fields := strings.Fields(line)
+		// If line indicates an interface
+		if len(fields) >= 2 && strings.HasSuffix(fields[0], ":") {
+			ifaceName = strings.Split(strings.TrimSuffix(fields[1], ":"), "@")[0]
+		}
+
+		// If the line contains an IP address (either IPv4 or IPv6)
+		if len(fields) > 1 && (fields[0] == "inet" || fields[0] == "inet6") {
+			ipAddress := fields[1]
+			// Add the IP address to the list for the corresponding interface
+			portsAndIPs[ifaceName] = append(portsAndIPs[ifaceName], ipAddress)
+		}
+	}
+	return portsAndIPs, nil
+}
+
+// checkIPReleased verifies if the VLAN IP from the deleted pods is released.
+func (data *testData) checkIPReleased(ipPoolName, podIPString string) error {
+	// Fetch the IPPool CR
+	ipPool, err := data.crdClient.CrdV1beta1().IPPools().Get(context.TODO(), ipPoolName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get IPPool %s: %v", ipPoolName, err)
+	}
+
+	podIP := net.ParseIP(podIPString)
+	if podIP == nil {
+		return fmt.Errorf("invalid pod IP: %s", podIPString)
+	}
+
+	// Check if podIP exists in the allocated IPs
+	for _, ipAddress := range ipPool.Status.IPAddresses {
+		if podIP.Equal(net.ParseIP(ipAddress.IPAddress)) {
+			return fmt.Errorf("IP is still allocated: %v", err)
+		}
+	}
+
+	return nil // IP is released
+}
+
+// getIPPoolNames fetches the IPPool names from a given NetworkAttachmentDefinition
+func (data *testData)getIPPoolNames(networkName string) ([]string, error) {
+	cmd := exec.Command("kubectl", "get", "net-attach-def", networkName, "-o", "jsonpath={.spec.config}")
```

Comment:
Better to create a netAttachDefClient and use it to get the NetAttachDef CR.

You may add a GetKubeConfig func to TestData to get the kubeconfig to create the client.